src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,257 @@
+
+_Bohr2Ang = unit_convert("Bohr", "Ang")
+_J2eV = unit_convert("J", "eV")
+_hc_eV = constant.h * constant.c * _J2eV
+
+@set_module("sisl.io.siesta")
+class vectorsSileSiesta(SileSiesta):
+    """Phonon eigenmode file"""
+
+    def _setup(self, *args, **kwargs):
+        """Simple setup that needs to be overwritten
+
+        All _r_next_* methods expect the fortran file unit to be handled
+        and that the position in the file is correct.
+        """
+        super()._setup(*args, **kwargs)
+
+        # default lattice
+        lattice = None
+
+        # In case the instantiation was called with wfsxSileSiesta("path", geometry=geometry)
+        parent = kwargs.get("parent")
+        if parent is None:
+            geometry = None
+        elif isinstance(parent, Geometry):
+            geometry = parent
+        elif isinstance(parent, Lattice):
+            lattice = parent
+        else:
+            geometry = parent.geometry
+
+        geometry = kwargs.get("geometry", geometry)
+        if geometry is not None and lattice is None:
+            lattice = geometry.lattice
+
+        lattice = kwargs.get("lattice", kwargs.get("sc", lattice))
+        if lattice is None and geometry is not None:
+            raise ValueError(
+                f"{self.__class__.__name__}(geometry=Geometry, lattice=None) is not an allowed argument combination."
+            )
+
+        if parent is None:
+            parent = geometry
+        if parent is None:
+            parent = lattice
+
+        self._parent = parent
+        self._geometry = geometry
+        self._lattice = lattice
+        if self._parent is None and self._geometry is None and self._lattice is None:
+
+            def conv(k):
+                if not np.allclose(k, 0.0):
+                    warn(
+                        f"{self.__class__.__name__} cannot convert stored k-points from 1/Ang to reduced coordinates. "
+                        "Please ensure parent=Hamiltonian, geometry=Geometry, or lattice=Lattice to ensure reduced k."
+                    )
+                return k / _Bohr2Ang
+
+        else:
+
+            def conv(k):
+                return (k @ lattice.cell.T) / (2 * np.pi * _Bohr2Ang)
+
+        self._convert_k = conv
+
+    def _open(self, rewind=False):
+        """Open the file
+
+        Here we also initialize some variables to keep track of the state of the read.
+        """
+        super()._open()
+        if rewind:
+            self.seek(0)
+        # Here we initialize the variables that will keep track of the state of the read.
+        # The process for identification is done on this basis:
+        #  _ik is the current (Python) index for the k-point to be read
+        #  _state is:
+        #        -1 : the file-descriptor has just been opened (i.e. in front of header)
+        #         0 : it means that the file-descriptor is in front of k point information
+        self._state = -1
+        self._ik = 0
```

Comment:
I don't know if this is even necessary, I guess the `read_eigenmode` could use the `SileBinder` method, in which case it isn't really necessary with all these details.

Reply:
yeah, not first priority, so disregard this comment. Thanks!